clients/src/main/java/org/apache/kafka/clients/admin/internals/AlterConsumerGroupOffsetsHandler.java

```diff
@@ -0,0 +1,156 @@
+import org.apache.kafka.common.message.OffsetCommitRequestData.OffsetCommitRequestTopic;
+import org.apache.kafka.common.message.OffsetCommitResponseData.OffsetCommitResponsePartition;
+import org.apache.kafka.common.message.OffsetCommitResponseData.OffsetCommitResponseTopic;
+import org.apache.kafka.common.protocol.Errors;
+import org.apache.kafka.common.requests.AbstractResponse;
+import org.apache.kafka.common.requests.OffsetCommitRequest;
+import org.apache.kafka.common.requests.OffsetCommitResponse;
+import org.apache.kafka.common.requests.FindCoordinatorRequest.CoordinatorType;
+import org.apache.kafka.common.utils.LogContext;
+import org.slf4j.Logger;
+
+public class AlterConsumerGroupOffsetsHandler implements AdminApiHandler<CoordinatorKey, Map<TopicPartition, Errors>> {
+
+    private final CoordinatorKey groupId;
+    private final Map<TopicPartition, OffsetAndMetadata> offsets;
+    private final Logger log;
+    private final AdminApiLookupStrategy<CoordinatorKey> lookupStrategy;
+
+    public AlterConsumerGroupOffsetsHandler(
+        String groupId,
+        Map<TopicPartition, OffsetAndMetadata> offsets,
+        LogContext logContext
+    ) {
+        this.groupId = CoordinatorKey.byGroupId(groupId);
+        this.offsets = offsets;
+        this.log = logContext.logger(AlterConsumerGroupOffsetsHandler.class);
+        this.lookupStrategy = new CoordinatorStrategy(CoordinatorType.GROUP, logContext);
+    }
+
+    @Override
+    public String apiName() {
+        return "offsetCommit";
+    }
+
+    @Override
+    public AdminApiLookupStrategy<CoordinatorKey> lookupStrategy() {
+        return lookupStrategy;
+    }
+
+    public static AdminApiFuture.SimpleAdminApiFuture<CoordinatorKey, Map<TopicPartition, Errors>> newFuture(
+            String groupId
+    ) {
+        return AdminApiFuture.forKeys(Collections.singleton(CoordinatorKey.byGroupId(groupId)));
+    }
+
+    @Override
+    public OffsetCommitRequest.Builder buildRequest(int brokerId, Set<CoordinatorKey> keys) {
+        List<OffsetCommitRequestTopic> topics = new ArrayList<>();
+        Map<String, List<OffsetCommitRequestPartition>> offsetData = new HashMap<>();
+        for (Map.Entry<TopicPartition, OffsetAndMetadata> entry : offsets.entrySet()) {
+            String topic = entry.getKey().topic();
+            OffsetAndMetadata oam = entry.getValue();
+            offsetData.compute(topic, (key, value) -> {
+                if (value == null) {
+                    value = new ArrayList<>();
+                }
+                OffsetCommitRequestPartition partition = new OffsetCommitRequestPartition()
+                        .setCommittedOffset(oam.offset())
+                        .setCommittedLeaderEpoch(oam.leaderEpoch().orElse(-1))
+                        .setCommittedMetadata(oam.metadata())
+                        .setPartitionIndex(entry.getKey().partition());
+                value.add(partition);
+                return value;
+            });
```

Comment:
Wouldn't this be clearer as a computeIfAbsent()?
```suggestion
            OffsetCommitRequestPartition partition = new OffsetCommitRequestPartition()
                    .setCommittedOffset(oam.offset())
                    .setCommittedLeaderEpoch(oam.leaderEpoch().orElse(-1))
                    .setCommittedMetadata(oam.metadata())
                    .setPartitionIndex(entry.getKey().partition());
            offsetData.computeIfAbsent(topic, key -> new ArrayList<>()).add(partition);
```